### ISSUE TITLE
Enable Authority discovery record signing

### ISSUE THREAD

**Issue by bkchr** [paritytech/polkadot#4499]
After https://github.com/paritytech/substrate/pull/10317 authority discovery supports signing of records using the node key. This ensures that not someone tries to spoof to be someone else. 

CC @wigy-opensource-developer

**Comment by wigy-opensource-developer** [comment]
@bkchr If I assessed it correctly, the v2 authority discovery record protocol is active since v0.9.15 and although we can just assume that, but that version is deployed on most validators. So the current task is just to enable strict checking, so v0.9.17 will enforce the newly introduced signatures.

Since bridge validators also use v2 for a while now, we can [enable strict checking](https://github.com/paritytech/polkadot/blob/c9f4932948854aed53b42b32751e8888779318f8/bridges/bin/rialto/node/src/service.rs#L535) on them, too.

**Comment by bkchr** [comment]
> @bkchr If I assessed it correctly, the v2 authority discovery record protocol is active since v0.9.15 and although we can just assume that, but that version is deployed on most validators. So the current task is just to enable strict checking, so v0.9.17 will enforce the newly introduced signatures.

Your message made me realize that this isn't maybe not the case. Could you please check that `polkadot-v0.9.15` branch in Substrate and check that it contains your commit?



> Since bridge validators also use v2 for a while now, we can [enable strict checking](https://github.com/paritytech/polkadot/blob/c9f4932948854aed53b42b32751e8888779318f8/bridges/bin/rialto/node/src/service.rs#L535) on them, too.

This code isn't maintained here in this repo and can be completely ignored. We only care about Polkadot. Tests nets also don't really need this signing of records.

**Comment by wigy-opensource-developer** [comment]
Yeah, v2 records are not part of 0.9.15-1 yet. substrate/dht.proto at e6fbbd5cdf72a5ed7fd65138072ed1f8a320a33d · paritytech/substrate - https://github.com/paritytech/substrate/blob/e6fbbd5cdf72a5ed7fd65138072ed1f8a320a33d/client/authority-discovery/src/worker/schema/dht.proto

**Comment by bkchr** [comment]
Okay, then we need to postpone this.
